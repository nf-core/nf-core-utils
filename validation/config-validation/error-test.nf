#!/usr/bin/env nextflow

/*
 * Error condition test for nf-core-utils plugin color formatting
 * 
 * This test specifically validates error conditions and color formatting
 * It is expected to fail with specific error messages to test the color functionality
 */

// Import plugin functions for configuration validation
include { checkProfileProvided } from 'plugin/nf-core-utils'

nextflow.enable.dsl = 2

workflow {
    
    log.info "=========================================="
    log.info "Color Formatting Error Test"
    log.info "=========================================="
    
    // This test is designed to fail to test error formatting
    log.info "Testing profile with trailing comma (expected to fail)"
    
    // Test with the monochrome_logs parameter from params
    def invalid_args = ['nextflow', 'run', 'main.nf', '-profile', 'docker,']
    checkProfileProvided(invalid_args, params.monochrome_logs ?: true)
}