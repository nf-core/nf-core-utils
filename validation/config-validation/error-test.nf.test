nextflow_pipeline {

    name "Test nf-core-utils color formatting error messages"
    script "../config-validation/error-test.nf"
    config "../config-validation/nextflow.config"

    test("Error output - Monochrome") {
        
        when {
            params {
                monochrome_logs = true
            }
        }

        then {
            assert workflow.failed
            assert workflow.exitStatus != 0
            
            // Should contain error message without color codes
            assert workflow.stdout.any { it.contains("Pipeline is starting") }
            
            // Check stdout for the actual error message (error appears in stdout in nf-test)
            def errorMessage = workflow.stdout.join('\n')
            assert errorMessage.contains("trailing comma")
            assert errorMessage.contains("ERROR")
            assert errorMessage.contains("HINT")
            
            // Should NOT contain ANSI color codes
            assert !errorMessage.contains('\033[')
            
            // Snapshot monochrome error output
            assert snapshot(
                workflow.stdout.findAll { it.contains("ERROR") || it.contains("HINT") }
            ).match()
        }
    }

    test("Error output - Color") {
        
        when {
            params {
                monochrome_logs = false
            }
        }

        then {
            assert workflow.failed
            assert workflow.exitStatus != 0
            
            // Should contain error message with color codes
            assert workflow.stdout.any { it.contains("Pipeline is starting") }
            
            // Check stdout for the actual error message (error appears in stdout in nf-test)
            def errorMessage = workflow.stdout.join('\n')
            assert errorMessage.contains("trailing comma")
            assert errorMessage.contains("ERROR")
            assert errorMessage.contains("HINT")
            
            // Note: Color codes may not be visible in test output due to nf-test capturing
            // The important thing is that the function accepts the parameter without errors
            // In real usage, colors would be displayed based on terminal capabilities
            
            // Snapshot color error output
            assert snapshot(
                workflow.stdout.findAll { it.contains("ERROR") || it.contains("HINT") }
            ).match()
        }
    }
}