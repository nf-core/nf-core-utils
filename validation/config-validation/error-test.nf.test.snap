{
    "Error output - Monochrome": {
        "content": [
            [
                "ERROR ~ ERROR ~ The `-profile` option cannot end with a trailing comma, please remove it and re-run the pipeline!",
                "HINT: A common mistake is to provide multiple values separated by spaces e.g. `-profile test, docker`."
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-09-01T18:53:13.431098377"
    },
    "Error output - Color": {
        "content": [
            [
                "ERROR ~ ERROR ~ The `-profile` option cannot end with a trailing comma, please remove it and re-run the pipeline!",
                "HINT: A common mistake is to provide multiple values separated by spaces e.g. `-profile test, docker`."
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-09-01T18:53:15.822989509"
    }
}